File: scratch/check_db.py
import psycopg2

def check_wms():
    conn = psycopg2.connect(
        host="localhost",
        port=56432,
        database="lms_wms_dev",
        user="postgres",
        password="postgres"
    )
    cursor = conn.cursor()
    tables = [
        "Warehouses", "Blocks", "Zones", "Bins", "InventoryItems", 
        "OutboundOrders", "InboundReceipts", "PutawayTasks", 
        "PickTasks", "ReplenishmentTasks", "CountTasks", 
        "OperatorProfiles", "OperatorRoleAssignments"
    ]
    print("--- WMS DATABASE ---")
    for t in tables:
        try:
            cursor.execute(f'SELECT COUNT(*) FROM "{t}"')
            count = cursor.fetchone()[0]
            print(f"{t}: {count}")
        except Exception as e:
            print(f"Error reading table {t}: {e}")
            conn.rollback()
    cursor.close()
    conn.close()

def check_oms():
    conn = psycopg2.connect(
        host="localhost",
        port=56432,
        database="lms_oms_dev",
        user="postgres",
        password="postgres"
    )
    cursor = conn.cursor()
    tables = ["Orders", "OrderLines"]
    print("\n--- OMS DATABASE ---")
    for t in tables:
        try:
            cursor.execute(f'SELECT COUNT(*) FROM "{t}"')
            count = cursor.fetchone()[0]
            print(f"{t}: {count}")
        except Exception as e:
            print(f"Error reading table {t}: {e}")
            conn.rollback()
    cursor.close()
    conn.close()

if __name__ == "__main__":
    check_wms()
    check_oms()

File: scratch/check_db.js
const { Client } = require('pg');
const client = new Client({
  user: 'postgres',
  host: '127.0.0.1',
  database: 'lms_wms_dev',
  password: 'postgres',
  port: 56432,
});

async function check() {
  await client.connect();
  console.log("Connected to WMS db");

  const whId = 'a3a1a1a1-a1a1-a1a1-a1a1-a1a1a1a1a1a1';
  const res = await client.query(`
    SELECT ora."Id", op."DisplayName", op."OperatorSub", r."Code" as "RoleCode", ora."Status"
    FROM "OperatorRoleAssignments" ora
    JOIN "operator_profiles" op ON ora."OperatorProfileId" = op."Id"
    JOIN "Roles" r ON ora."RoleId" = r."Id"
    WHERE ora."WarehouseId" = $1
  `, [whId]);

  console.log(`Role Assignments in HCM Mega Hub (${whId}):`);
  res.rows.forEach(r => {
    console.log(`- AssignId: ${r.Id}, Name: ${r.DisplayName}, Sub: ${r.OperatorSub}, Role: ${r.RoleCode}, Status: ${r.Status}`);
  });

  await client.end();
}

check().catch(err => {
  console.error("DB Check Error:", err);
  process.exit(1);
});
